concordium-std/src/impls.rs

```diff
@@ -1640,6 +1640,24 @@ impl HasPolicy for Policy<AttributesCursor> {
         self.items.remaining_items -= 1;
         Some((AttributeTag(tag_value_len[0]), tag_value_len[1]))
     }
+
+    fn iter(&mut self) -> PolicyAttributesIter<Self> {
+        PolicyAttributesIter {
+            iter: self,
+            buf:  [0u8; 31],
+        }
+    }
+}
+
+impl<P: HasPolicy> Iterator for PolicyAttributesIter<'_, P> {
```

Comment:
You should be able to implement [`ExactSizeIterator`](https://doc.rust-lang.org/std/iter/trait.ExactSizeIterator.html) for `PolicyAttributesIter` and [`IntoIterator`](https://doc.rust-lang.org/std/iter/trait.IntoIterator.html) for `HasPolicy` as well

Reply:
I can't implement `IntoIterator` for `P: HasPolicy` due to the orphan rule. But I implemented `ExactSizeIterator` by adding an `remaining_items` function toe `HasPolicy`.